conan/tools/cmake/toolchain/blocks.py

```diff
@@ -112,6 +112,53 @@ def context(self):
 
         return {"vs_runtimes": config_dict}
 
+class VSDebuggerEnvironment(Block):
+    template = textwrap.dedent("""
+        # Definition of CMAKE_VS_DEBUGGER_ENVIRONMENT
+        {% if vs_debugger_path %}
+        {% set genexpr = namespace(str='') %}
+        {% for config, value in vs_debugger_path.items() %}
+        {% set genexpr.str = genexpr.str +
+                                '$<$<CONFIG:' + config + '>:' + value|string + '>' %}
+        {% endfor %}
+        set(CMAKE_VS_DEBUGGER_ENVIRONMENT "PATH={{ genexpr.str }};%PATH%")                      
+        {% endif %}
+        """)
+    
+    def context(self):
+        os_ = self._conanfile.settings.get_safe("os")
+        build_type = self._conanfile.settings.get_safe("build_type")
+
+        if (os_ and not "Windows" in os_) or not build_type:
+            return None
+        
+        generator = self._conanfile.conf.get("tools.cmake.cmaketoolchain:generator", default="Visual Studio")
+        if not "Visual Studio" in generator:
+            return None
```

Comment:
You can check ``"Visual" not in self._toolchain.generator:``, the ``toolchain.generator`` already contains the generator defined

Reply:
Done!